es-app/src/guis/GuiSlideshowScreensaverOptions.cpp

```diff
@@ -0,0 +1,122 @@
+	addSaveFunc([sss_bg_audio_file] {
+		Settings::getInstance()->setString("SlideshowScreenSaverBackgroundAudioFile", sss_bg_audio_file->getValue());
+	});
+
+	// image source
+	auto sss_source = std::make_shared< OptionListComponent<std::string> >(mWindow, "IMAGE SOURCE", false);
+	std::vector<std::string> image_sources;
+	image_sources.push_back("game list");
+	image_sources.push_back("custom");
+	for(auto it = image_sources.begin(); it != image_sources.end(); it++)
+		sss_source->add(*it, *it, Settings::getInstance()->getString("SlideshowScreenSaverImageSource") == *it);
+	addWithLabel(row, "IMAGE SOURCE", sss_source);
+	addSaveFunc([this, sss_source] {
+		Settings::getInstance()->setString("SlideshowScreenSaverImageSource", sss_source->getSelected());
+	});
+
+	// custom image directory
+	auto sss_image_dir = std::make_shared<TextComponent>(mWindow);
+	addEditableTextComponent(row, "CUSTOM IMAGE DIR", sss_image_dir, Settings::getInstance()->getString("SlideshowScreenSaverImageDir"));
+	addSaveFunc([sss_image_dir] {
+		Settings::getInstance()->setString("SlideshowScreenSaverImageDir", sss_image_dir->getValue());
+	});
+
+	// recurse custom image directory
+	auto sss_recurse = std::make_shared<SwitchComponent>(mWindow);
+	sss_recurse->setState(Settings::getInstance()->getBool("SlideshowScreenSaverRecurse"));
+	addWithLabel(row, "CUSTOM IMAGE DIR RECURSIVE", sss_recurse);
+	addSaveFunc([sss_recurse] {
+		Settings::getInstance()->setBool("SlideshowScreenSaverRecurse", sss_recurse->getState());
+	});
+
+	// custom image filter
+	auto sss_image_filter = std::make_shared<TextComponent>(mWindow);
+	addEditableTextComponent(row, "CUSTOM IMAGE FILTER", sss_image_filter, Settings::getInstance()->getString("SlideshowScreenSaverImageFilter"));
+	addSaveFunc([sss_image_filter] {
+		Settings::getInstance()->setString("SlideshowScreenSaverImageFilter", sss_image_filter->getValue());
+	});
+}
+
+GuiSlideshowScreensaverOptions::~GuiSlideshowScreensaverOptions()
+{
+}
+
+void GuiSlideshowScreensaverOptions::addWithLabel(ComponentListRow row, const std::string label, std::shared_ptr<GuiComponent> component)
+{
+	row.elements.clear();
+
+	auto lbl = std::make_shared<TextComponent>(mWindow, strToUpper(label), Font::get(FONT_SIZE_MEDIUM), 0x777777FF);
+	row.addElement(lbl, true); // label
+
+	row.addElement(component, false, true);
+
+	addRow(row);
+}
+
+void GuiSlideshowScreensaverOptions::addEditableTextComponent(ComponentListRow row, const std::string label, std::shared_ptr<GuiComponent> ed, std::string value)
+{
+	row.elements.clear();
+
+	auto lbl = std::make_shared<TextComponent>(mWindow, strToUpper(label), Font::get(FONT_SIZE_MEDIUM), 0x777777FF);
+	row.addElement(lbl, true); // label
+
+	//std::shared_ptr<GuiComponent> ed = std::make_shared<TextComponent>(window, "", Font::get(FONT_SIZE_SMALL, FONT_PATH_LIGHT), 0x777777FF, ALIGN_RIGHT);
+	row.addElement(ed, true);
```

Comment:
Here ed would be null or am I missing something? Also remove your commented code if it will not be used.